cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -296,6 +303,49 @@ where
 			.ok_or(sp_blockchain::Error::Backend(format!("Header not found for {:?}", at)))
 	}
 
+	fn block_body(&self, at: Block::Hash) -> Result<Vec<Block::Extrinsic>, sp_blockchain::Error> {
```

Comment:
These methods really trigger me. They are trivial. but leave mixed impression. This for for instance indicates that `Executor` has ability to extract block body from somewhere, while that is not the case at all, block body is extracted from the client.

Reply:
Client is part of `Executor`, so I don't see why it's unfair to say Executor has the ability to retrieve the block body...

Reply:
Conceptually it is not Executor's ability, hence API is kind of misleading and convenience benefit is quite low IMHO

Reply:
Hmm, Executor is still a full node that has the ability to retrieve the recent blocks for sure, but it makes sense if you were thinking that in the ideal world Executor even doesn't have to store the recent blocks and have some other way to fetch the required extrinsics. Given that this helper method has a similar pattern to `Executor.header()`, I just extracted both of them. Currently, it has been used only two times, I'm ok with removing it if that bothers you.

Reply:
Executor as a node - sure, but that is why we have `Client` API. I just don't think we need to expose the same capabilities from `sc-cirrus-executor`'s `Executor` struct and other structs, because it is not their responsibility to get blocks.